### PR TITLE
Add demo and doc for memory viewer.

### PR DESCRIPTION
**Pull request by qiuminxu**
Also added minor tweaks to better align the peak memory allocation band.

Screenshot of the demo:
memory-viewer-demo

### CODE REVIEW

File: tensorboard/plugins/profile/README.md [hunk 1]
```diff
@@ -97,3 +97,33 @@ The analysis contains three sections:
   * Host-side analysis, which shows you the detailed analysis on the host side,
   including a breakdown of input processing time on the host, and a tabular
   view of details for each input operation.
+
+## Memory Viewer
+
+Memory viewer tool allows you to visualize the peak memory usage for your
+program, and how the memory usage trends over the program's lifetime.
+
+![Memory_Viewer](docs/memory-viewer.png)
```
Comment (jart) [inline tensorboard/plugins/profile/README.md]: [Optional] Probably not worth the time to change, but `sudo apt-get install pngcrush` is an amazingly good program for files like these.

File: tensorboard/plugins/profile/README.md [hunk 2]
```diff
@@ -97,3 +97,33 @@ The analysis contains three sections:
...
+
+The top-most display shows how many memory usage trends over the program's
+lifetime. The band drawn from top to buttom of the plot indicates the point
+in the program with peak memory utilization.
+
+Each point on the plot represents a "program point" in XLA's Hige Level
```
Comment (jart) [inline tensorboard/plugins/profile/README.md]: Typo on 'High'?

File: tensorboard/plugins/profile/README.md (hunk 1 above)
Comment (qiuminxu) [reply tensorboard/plugins/profile/README.md]: Thanks for the suggestion, it is indeed a great tool.

**Comment by qiuminxu** [comment]
The typo is also fixed.

**Pull request merged**